dd-java-agent/instrumentation/spring-data/src/main/java/datadog/trace/instrumentation/springdata/SpringRepositoryInstrumentation.java

```
@@ -0,0 +1,118 @@
+      repositoryFactorySupport.addRepositoryProxyPostProcessor(
+          InterceptingRepositoryProxyPostProcessor.INSTANCE);
+    }
+  }
+
+  public static final class InterceptingRepositoryProxyPostProcessor
+      implements RepositoryProxyPostProcessor {
+    public static final RepositoryProxyPostProcessor INSTANCE =
+        new InterceptingRepositoryProxyPostProcessor();
+
+    // DQH - TODO: Support older versions?
+    // The signature of postProcess changed to add RepositoryInformation in
+    // spring-data-commons 1.9.0
+    // public void postProcess(final ProxyFactory factory) {
+    //   factory.addAdvice(0, RepositoryInterceptor.INSTANCE);
+    // }
+
+    @Override
+    public void postProcess(
+        final ProxyFactory factory, final RepositoryInformation repositoryInformation) {
+      factory.addAdvice(0, RepositoryInterceptor.INSTANCE);
+    }
+  }
+
+  static final class RepositoryInterceptor implements MethodInterceptor {
+    public static final MethodInterceptor INSTANCE = new RepositoryInterceptor();
+
+    private RepositoryInterceptor() {}
+
+    @Override
+    public Object invoke(final MethodInvocation methodInvocation) throws Throwable {
+      final Method invokedMethod = methodInvocation.getMethod();
+      final Class<?> clazz = invokedMethod.getDeclaringClass();
+
+      boolean isRepositoryOp = Repository.class.isAssignableFrom(clazz);
+      // non-Repository methods including Object methods will also flow through
+      // this interceptor.  Don't create spans for those.
+      if (!isRepositoryOp) {
+        return methodInvocation.proceed();
+      }
+
+      final Scope scope = GlobalTracer.get().buildSpan("repository.operation").startActive(true);
+      final Span span = scope.span();
+      DECORATOR.afterStart(span);
+      DECORATOR.onOperation(span, invokedMethod);
+
+      Object result = null;
+      try {
+        result = methodInvocation.proceed();
+      } catch (Throwable t) {
```

Just curious...
- Do we want to catch & propagate?  Just catch?  
- Catch Throwable or just Exception?

---

If the exception was not caused by us (ie, it is generated by their code), then we should rethrow the exception.  We don't want to modify the regular behavior of the application.

---

Okay, that's what I would expect, but it seem entirely consistent with what I saw elsewhere.